client/src/api/v2/camera.ts

```diff
@@ -0,0 +1,271 @@
+
+enum CameraDevice {
+  Primary = 'primary',
+  Secondary = 'secondary',
+}
+
+interface CameraOverlayStatus {
+  /** List of available overlays */
+  overlays: string[];
+  /** Name of currently active overlay */
+  active: string;
+}
+
+/**
+ * Use a list of camera overlays
+ *
+ * @param device Camera screen
+ * @returns Status of overlays and function to set active overlay.
+ */
+export function useOverlays(device: CameraDevice) {
+  const [overlays, setOverlays] = useState<CameraOverlayStatus | null>(null);
+
+  const handleData = useCallback(
+    (data) => {
+      const json = JSON.parse(data);
+      if (json.device === device) {
+        setOverlays({
+          active: json.activeOverlay,
+          overlays: json.overlays,
+        });
+      }
+    },
+    [device],
+  );
+  useChannel('push-overlays', handleData);
+
+  useEffect(() => {
+    emit('get-overlays');
+  }, [device]);
+
+  const setActiveOverlay = useCallback(
+    (activeOverlay) => {
+      emit(
+        'set-overlays',
+          JSON.stringify({
+            [device]: activeOverlay,
+          }),
+      );
+    },
+    [device],
+  );
+
+  return { overlays, setActiveOverlay };
+}
+
+/**
+ * Send a message to server to start recording video
+ */
+export function startRecording() {
+  emit('start-camera-recording');
+}
+
+/**
+ * Send a message to server to stop recording video
+ */
+export function stopRecording() {
+  emit('stop-camera-recording');
+}
+
+enum CameraRecordingStatus {
+  Off = 'off',
+  Recording = 'recording',
+  Error = 'error',
+}
+
+interface CameraRecordingStatusPayload {
+  /** Status of the camera's recording */
+  status: CameraRecordingStatus;
+  /** Current duration of recording in minutes, if recording */
+  recordingMinutes?: number;
+  /** Filename of recording, if recording */
+  recordingFile?: string;
+  /** Remaining disk space available for recording, in bytes */
+  diskSpaceRemaining: number;
+  /** Description of any error that has occurred */
+  error?: string;
+  /** Catch-all for any extra data */
+  [propName: string]: any;
+}
+
+interface CameraRecordingStatusFormatted {
+  /** Status of the camera's recording */
+  status?: string;
+  /** Current duration of recording in mm:ss, if recording */
+  recordingMinutes?: string;
+  /** Filename of recording, if recording */
+  recordingFile?: string;
+  /** Remaining disk space available for recording */
+  diskSpaceRemaining?: string;
+  /** Description of any error that has occurred */
+  error?: string;
+  /** Catch-all for any extra data */
+  [propName: string]: any;
+}
+
+/**
+ * Parse the recording payload into an object and format the values
+ *
+ * Payload structure is defined in the 'V3 MQTT Topics' page on Notion.
+ * Topic is /v3/camera/recording/status/<primary/>secondary>.
+ *
+ * @param data Payload from MQTT message, parsed
+ * @returns Formatted payload
+ */
+function parseRecordingPayloadData(data: CameraRecordingStatusPayload) {
```

Comment:
Since data is nullable, it's good to annotate the type explicitly as "CameraRecordingStatusPayload?"